apps/angular/crud/src/app/app.component.ts

```diff
@@ -1,51 +1,52 @@
 import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
-import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
-
+import { Component, OnInit, effect } from '@angular/core';
+import { TodoService } from './services/todo.service';
+import { todo } from './todo.model';
+import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, MatProgressSpinnerModule],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
+    <!-- 
+    // this is optional  
+    @if (isError) {
+      <p class="error">{{ isError }}</p>
+    } -->
+    @if (isLoading) {
+      <mat-spinner></mat-spinner>
+    }
+    <div *ngFor="let todo of todos$">
       {{ todo.title }}
       <button (click)="update(todo)">Update</button>
+      <button (click)="delete(todo.id)">Delete</button>
     </div>
   `,
   styles: [],
 })
 export class AppComponent implements OnInit {
-  todos!: any[];
-
-  constructor(private http: HttpClient) {}
+  todos$!: todo[];
+  // isError!: string | null;
+  isLoading!: boolean;
 
+  constructor(private todoService: TodoService) {
+    effect(() => {
+      // const { todos, isError, isLoading } = this.todoService.appSignal();
+      const { todos, isLoading } = this.todoService.appSignal();
+      this.todos$ = todos;
+      // this.isError = isError;
+      this.isLoading = isLoading;
+    });
```

Comment:
I have added effect so that the value changes of signals so it will update the values of todos$. but now got it, so we can use computed signals and declared Signal at starting assigned values.


Thank you for your valuable review.